pkg/syncer/fullsync.go

```diff
@@ -442,6 +445,34 @@ func volumeInfoCRFullSync(ctx context.Context, metadataSyncer *metadataSyncInfor
 		}
 	}
 
+	cnsVolumeMap := make(map[string]cnstypes.CnsVolume)
+	if isStorageQuotaM2FSSEnabled {
+		var volManager volumes.Manager
+		if !isMultiVCenterFssEnabled {
+			volManager = MetadataSyncer.volumeManager
+		} else {
+			vCenter := MetadataSyncer.configInfo.Cfg.Global.VCenterIP
+			cnsVolumeMgr, volMgrFound := MetadataSyncer.volumeManagers[vCenter]
+			if !volMgrFound {
+				log.Errorf("could not get volume manager for the vCenter: %q", vCenter)
+				return
+			}
+			volManager = cnsVolumeMgr
+		}
+
+		queryFilter := cnstypes.CnsQueryFilter{
+			ContainerClusterIds: []string{
+				MetadataSyncer.configInfo.Cfg.Global.ClusterID,
+			},
+		}
+		queryAllResult, err := volManager.QueryAllVolume(ctx, queryFilter, cnstypes.CnsQuerySelection{})
```

Comment:
THey already do a QueryVolume at https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/fullsync.go#L146 , move that before the volumeInfoCRFullSync and pass that to volumeInfoCRFullSync as a parameter